app/src/marketplace/utils/auth.js

```diff
@@ -23,12 +36,13 @@ export const userIsNotAuthenticated = connectedRouterRedirect({
     authenticatingSelector,
     // This sends the user either to the query param route if we have one, or to
     // the landing page if none is specified and the user is already logged in
-    redirectPath: (state, ownProps) => locationHelper.getRedirectQueryParam(ownProps) || '/',
+    redirectPath: (state, ownProps) => locationHelper.getRedirectQueryParam(ownProps) || routes.streams(),
     // This prevents us from adding the query parameter when we send the user away from the login page
     allowRedirectBack: false,
     // If selector is true, wrapper will not redirect
     // So if there is no user data, then we show the page
     authenticatedSelector: (state) => !selectUserData(state),
     // A nice display name for this check
     wrapperDisplayName: 'UserIsNotAuthenticated',
+    FailureComponent,
```

Comment:
Here's an interesting story.

`redux-auth-wrapper` does the "successful login" redirection. There's no extra logic for it beyond `userIsNotAuthenticated`, meaning, on success we hit `getUserData` and redirections along with respecting `redirect` from the URL is taken care of organically.

Now there are 2 scenarios:

First in which `redirect` is a path to a protected page. Here's how it goes:
1. `redux-auth-wrapper` mounts `FailureComponent`
2. `FailureComponent` redirects to the path w/o further failures
3. Protected page gets loaded (`FailureComponent` is gone at this point)
4. All good!

Second scenario in which `redirect` is a path to a page that requires user to not be signed in, let's say `/login`: 
1. `redux-auth-wrapper` mounts `FailureComponent`
2. `FailureComponent` redirects to `/login` and fails again
3. `FailureComponent` gets rerendered (still mounted) and DOES NOT redirect cause the original `FailureComponent` doesn't implement `componentDidUpdate`.
4. `null` is for blank page.

**tl;dr** We needed a better `FailureComponent`, one that knows how to handle updates.